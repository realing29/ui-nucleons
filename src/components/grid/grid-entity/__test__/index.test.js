import React from 'react';
import { shallow } from 'enzyme/build';
import GridEntity from '../';

describe('<GridEntity/>', () => {
  it('should renders correctly without props', () => {
    const gridEntity = shallow(<GridEntity />);
    expect(gridEntity).toMatchSnapshot();
  });
  it('should renders correctly with props', () => {
    const gridEntity = shallow(
      <GridEntity createClasses={jest.fn(() => 'test classes')} tag='article'>
        <div>children</div>
      </GridEntity>
    );
    expect(gridEntity).toMatchSnapshot();
  });
  it('should renders correctly with different tags', () => {
    let gridEntity = shallow(<GridEntity />);
    expect(gridEntity.find('div')).toHaveLength(1);
    gridEntity = shallow(<GridEntity tag='section' />);
    expect(gridEntity.find('section')).toHaveLength(1);
    gridEntity = shallow(<GridEntity tag='p' />);
    expect(gridEntity.find('p')).toHaveLength(1);
  });
  it('should renders correctly with createClasses function', () => {
    const createClasses = jest.fn();
    shallow(<GridEntity createClasses={createClasses} />);
    expect(createClasses).toHaveBeenCalledTimes(1);
  });
});
