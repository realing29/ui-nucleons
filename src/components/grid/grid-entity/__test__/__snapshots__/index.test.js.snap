// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GridEntity/> should renders correctly with props 1`] = `
<article
  className="test classes"
>
  <div>
    children
  </div>
</article>
`;

exports[`<GridEntity/> should renders correctly without props 1`] = `
<div
  className={null}
/>
`;
